driver/k8s/node_service.go

```diff
@@ -0,0 +1,103 @@
+package k8s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/cybozu-go/topolvm"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// NodeService represents node service.
+type NodeService struct {
+	client.Client
+}
+
+// NewNodeService returns NodeService.
+func NewNodeService(mgr manager.Manager) *NodeService {
+	return &NodeService{Client: mgr.GetClient()}
+}
+
+func (s NodeService) getNodes(ctx context.Context) (*corev1.NodeList, error) {
+	nl := new(corev1.NodeList)
+	err := s.List(ctx, nl)
+	if err != nil {
+		return nil, err
+	}
+	return nl, nil
+}
+
+func (s NodeService) extractCapacityFromAnnotation(node *corev1.Node) (int64, error) {
+	c, ok := node.Annotations[topolvm.CapacityKey]
+	if !ok {
+		return 0, fmt.Errorf("%s is not found", topolvm.CapacityKey)
+	}
+	return strconv.ParseInt(c, 10, 64)
+}
+
+// GetCapacityByName returns VG capacity of specified node by name.
+func (s NodeService) GetCapacityByName(ctx context.Context, name string) (int64, error) {
+	n := new(corev1.Node)
+	err := s.Get(ctx, client.ObjectKey{Name: name}, n)
+	if err != nil {
+		return 0, err
+	}
+
+	return s.extractCapacityFromAnnotation(n)
+}
+
+// GetCapacityByNodeNumber returns VG capacity of specified node by node number.
+func (s NodeService) GetCapacityByNodeNumber(ctx context.Context, requestNodeNumber string) (int64, error) {
```

Comment:
"Node number" is not defined.  Therefore, the intention of this method is vague.